modules/dlm/src/test/java/org/elasticsearch/dlm/ModifyDataLifecycleServiceTests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.dlm;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.DataLifecycle;
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.List;
+
+import static org.elasticsearch.dlm.DLMFixtures.createDataStream;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.Mockito.mock;
+
+public class ModifyDataLifecycleServiceTests extends ESTestCase {
+
+    private final ModifyDataLifecycleService service = new ModifyDataLifecycleService(mock(ClusterService.class));
```

Comment:
We can use a real `ClusterService` instead of a mock.

See
```
org.elasticsearch.test.ClusterServiceUtils.createClusterService
```
and
```
org.elasticsearch.test.ClusterServiceUtils.setState
```

Reply:
On it.

Reply:
Since we moved this to `MetadataDataStreamsService` I followed the same convention we had for the other modify methods and made it a static method.